src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs

```diff
@@ -1,24 +1,21 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Immutable;
-
 namespace Microsoft.CodeAnalysis.Semantics
 {
-    public enum SimpleUnaryOperationKind
+    /// <summary>
+    /// Kind of unary operator
+    /// </summary>
+    public enum UnaryOperatorKind
     {
         None = 0x0,
 
         BitwiseNegation = 0x1,
         LogicalNot = 0x2,
-        PostfixIncrement = 0x3,
-        PostfixDecrement = 0x4,
-        PrefixIncrement = 0x5,
-        PrefixDecrement = 0x6,
-        Plus = 0x7,
-        Minus = 0x8,
-        True = 0x9,
-        False = 0xa,
-        BitwiseOrLogicalNot = 0xb,
+        Plus = 0x3,
+        Minus = 0x4,
+        True = 0x5,
+        False = 0x6,
+        BitwiseOrLogicalNot = 0x7,
```

Comment:
Actually, I looked at the other logical operators in VB 'And', 'Or' and 'Xor', and they have also similar differing semantics based on the type of the operands, but we use a single BinaryOperatorKind to represent them. I recall that the design team has decided that we will not have different OperatorKind for the same operator just due to different semantics based on operand types. So, I am going to delete BitwiseOrLogicalNot and we will use 'LogicalNot' to represent VB 'Not' operator and C# '!' operator. 'BitwiseNegation' will be used only for the C# '~' operator.

---
In reply to: [134536171](https://github.com/dotnet/roslyn/pull/21407#discussion_r134536171) [](ancestors = 134536171,134530885)